openlibrary/plugins/admin/code.py

```diff
@@ -201,6 +205,56 @@ def POST(self):
         return delegate.RawText(json.dumps(results), content_type="application/json")
 
 
+class resolve_redirects:
+    def GET(self):
+        params = web.input(key='', test=False)
+        summary = {
+            'key': params.key,
+            'redirect_chain': [],
+            'resolved_key': None
+        }
+        if params.key:
+            key = params.key
+            while not summary['resolved_key']:
+                thing = web.ctx.site.get(key)
+                if thing.type.key == "/type/redirect":
+                    summary['redirect_chain'].append({
+                        "key": thing.key,
+                        "occurrences": {},
+                        "updates": {},
+                    })
+                    key = thing.location
+                else:
+                    summary['resolved_key'] = thing.key
```

Comment:
Might be best to pull out into a function like `def find_redirect_chain(key: str) -> list[str]` return just the `keys`

```py
summary['redirect_chain'] = [
    { 'key': k, 'occurrences': {}, 'updates': {} }
    for k in first_redirect_chain(key)
]
```